### PR TITLE
rpc: Change RPCExamples to bech32

### LINKED ISSUE #18185: Change address format in RPCExamples from legacy to bech32
Regarding that Bitcoin Core switched to bech32 as the default format for addresses recently (with release 0.19.0.1), this should also be reflected in the RPC help texts. The change was already done for the RPCs `getaddressinfo` (commit https://github.com/bitcoin/bitcoin/commit/2ee0cb3330ccf70f0540cb42370796e32eff1569 by jonatack, PR https://github.com/bitcoin/bitcoin/pull/17283) and `validateaddress` (commit https://github.com/bitcoin/bitcoin/pull/18122/commits/7f1475c7119e8c72bce39a63386a6ca859066b80 by my humble self, PR https://github.com/bitcoin/bitcoin/pull/18122/), but there are still some RPCExamples around with Base58 addresses, i.e. 
- `setlabel`
- `sendtoaddress`
- `getreceivedbyaddress`
- .....

All those legacy address usages should be identified and substituted by bech32 addresses where it makes sense (note that e.g. for `signmessage` and `verifymessage` there is as far as I know no possibility to sign with segwit addresses yet, hence the legacy addresses should stay). For the address, the newly introduced string constant `EXAMPLE_ADDRESS` (in `src/rpc/util.h`) can be used, which contains an illegal bech32 address.

#### Useful skills:

Basic understanding of address formats.

#### Want to work on this issue?

The purpose of the `good first issue` label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.

You do not need to request permission to start working on this. You are encouraged to comment on the issue if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by yusufsahinhamza**
This is a follow-up PR to #18197 that fixes RPCExamples.

Fixes #18185.

### CODE REVIEW

**Comment by leto** [comment]
utACK: looks like an improvement to the tests + docs

**Comment by Empact** [comment]
Concept ACK

File: doc/developer-notes.md [hunk 1]
```diff
@@ -1089,11 +1089,11 @@ A few guidelines for introducing and reviewing new RPC interfaces:
     new RPC is replacing a deprecated RPC, to avoid both RPCs confusingly
     showing up in the command list.
 
-- Use *invalid* bech32 addresses (e.g. the constant `EXAMPLE_ADDRESS`) for
+- Use *invalid* addresses (e.g. in the constant array `EXAMPLE_ADDRESS`) for
```
Comment (theStack) [inline doc/developer-notes.md]: The "bech32" part is actually important here, as we want to get rid of legacy addresses in the long term. Without that the reader of the guideline may think that any address format is fine.

File: doc/developer-notes.md [hunk 2]
```diff
@@ -1089,11 +1089,11 @@ A few guidelines for introducing and reviewing new RPC interfaces:
...
   `RPCExamples` help documentation.
 
   - *Rationale*: Prevent accidental transactions by users and encourage the use
-    of bech32 addresses by default.
+    of example addresses by default.
```
Comment (theStack) [inline doc/developer-notes.md]: That doesn't make sense. "Encourage the use of..." adresses the _user_ of the software here, not the developer. As above, "bech32" should stay here.

File: src/rpc/misc.cpp [hunk 3]
```diff
@@ -44,8 +44,8 @@ static UniValue validateaddress(const JSONRPCRequest& request)
             "}\n"
                 },
                 RPCExamples{
-                    HelpExampleCli("validateaddress", EXAMPLE_ADDRESS) +
-                    HelpExampleRpc("validateaddress", EXAMPLE_ADDRESS)
+                    HelpExampleCli("validateaddress", EXAMPLE_ADDRESS[1]) +
```
Comment (theStack) [inline src/rpc/misc.cpp]: The quotation marks around the address are missing. Previously they were included in `EXAMPLE_ADDRESS`, now as the constant contains the bare address, so they have to be added here.

File: src/wallet/rpcwallet.cpp [hunk 4]
```diff
@@ -3774,8 +3774,8 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
             "}\n"
                 },
                 RPCExamples{
-                    HelpExampleCli("getaddressinfo", EXAMPLE_ADDRESS) +
-                    HelpExampleRpc("getaddressinfo", EXAMPLE_ADDRESS)
+                    HelpExampleCli("getaddressinfo", EXAMPLE_ADDRESS[1]) +
```
Comment (theStack) [inline src/wallet/rpcwallet.cpp]: The quotation marks around the address are missing (see comment to `"validateaddress"` above).

**Review by theStack — CHANGES_REQUESTED** [review]
I'm not sure if it's a good idea to mix legacy and bech32 addresses in the same examples address array. The confusing consequence is that the index of the address array determines the resulting address type, and if someone e.g. would add another legacy address in front, many help strings that accessed at index 1 to get bech32 examples address woud then show legacy addresses again. To prevent that, it would be even okay right now if the legacy addresses stay in hardcoded (and the guidelines remain untouched).

**Comment by theStack** [comment]
LGTM now, can you squash the commits, please? In the course of that, you could also change the commit title to be more descriptive (e.g. "change to bech32" than just "fix").

**Review by jonatack — COMMENTED** [review]
Concept ACK

**Comment by MarcoFalke** [comment]
ACK 3e32499909ca8127baaa9b40ad113b25ee151bbd

**Review by jonatack — COMMENTED** [review]
ACK 3e32499

**Pull request merged**